cmd/run.go

```diff
@@ -39,88 +39,54 @@ const (
-	cmd.PersistentFlags().Int(flagMapping.LoaderHttpRetryDelay, defaultHttpRetryDelay, "http loader: The initial delay between retries in seconds")
-	cmd.PersistentFlags().String(flagMapping.LoaderFilePath, "config.yaml", "file loader: The path to the file to read the runtime config from")
-	cmd.PersistentFlags().String(flagMapping.TargetManagerConfig, "", "target manager: The path to the file to read the target manager config from")
-
-	_ = viper.BindPFlag(flagMapping.ApiAddress, cmd.PersistentFlags().Lookup(flagMapping.ApiAddress))
-	_ = viper.BindPFlag(flagMapping.SparrowName, cmd.PersistentFlags().Lookup(flagMapping.SparrowName))
-	_ = viper.BindPFlag(flagMapping.LoaderType, cmd.PersistentFlags().Lookup(flagMapping.LoaderType))
-	_ = viper.BindPFlag(flagMapping.LoaderInterval, cmd.PersistentFlags().Lookup(flagMapping.LoaderInterval))
-	_ = viper.BindPFlag(flagMapping.LoaderHttpUrl, cmd.PersistentFlags().Lookup(flagMapping.LoaderHttpUrl))
-	_ = viper.BindPFlag(flagMapping.LoaderHttpToken, cmd.PersistentFlags().Lookup(flagMapping.LoaderHttpToken))
-	_ = viper.BindPFlag(flagMapping.LoaderHttpTimeout, cmd.PersistentFlags().Lookup(flagMapping.LoaderHttpTimeout))
-	_ = viper.BindPFlag(flagMapping.LoaderHttpRetryCount, cmd.PersistentFlags().Lookup(flagMapping.LoaderHttpRetryCount))
-	_ = viper.BindPFlag(flagMapping.LoaderHttpRetryDelay, cmd.PersistentFlags().Lookup(flagMapping.LoaderHttpRetryDelay))
-	_ = viper.BindPFlag(flagMapping.LoaderFilePath, cmd.PersistentFlags().Lookup(flagMapping.LoaderFilePath))
-	_ = viper.BindPFlag(flagMapping.TargetManagerConfig, cmd.PersistentFlags().Lookup(flagMapping.TargetManagerConfig))
+	NewFlag("api.address", "apiAddress").String().Bind(cmd, ":8080", "api: The address the server is listening on")
+	NewFlag("name", "sparrowName").String().Bind(cmd, "", "The DNS name of the sparrow")
+	NewFlag("loader.type", "loaderType").StringP("l").Bind(cmd, "http", "Defines the loader type that will load the checks configuration during the runtime. The fallback is the fileLoader")
+	NewFlag("loader.interval", "loaderInterval").Int().Bind(cmd, defaultLoaderInterval, "defines the interval the loader reloads the configuration in seconds")
+	NewFlag("loader.http.url", "loaderHttpUrl").String().Bind(cmd, "", "http loader: The url where to get the remote configuration")
+	NewFlag("loader.http.token", "loaderHttpToken").String().Bind(cmd, "", "http loader: Bearer token to authenticate the http endpoint")
+	NewFlag("loader.http.timeout", "loaderHttpTimeout").Int().Bind(cmd, defaultLoaderHttpTimeout, "http loader: The timeout for the http request in seconds")
+	NewFlag("loader.http.retry.count", "loaderHttpRetryCount").Int().Bind(cmd, defaultHttpRetryCount, "http loader: Amount of retries trying to load the configuration")
+	NewFlag("loader.http.retry.delay", "loaderHttpRetryDelay").Int().Bind(cmd, defaultHttpRetryDelay, "http loader: The initial delay between retries in seconds")
```

Comment:
We still have an issue with the time/delay config here.
The default values do not fit anymore. Eg. defaultLoaderInterval is 300 -> 300ns and should be 300s.
Currently, we are not able to set a default of 300s since this would be an int=300000000000. This is not working for int32.
Any ideas? Should we switch it to type string for time/delay configs?

Reply:
Good catch I think we can just convert the `time.Duration` into int like this:
```go
const (
	defaultLoaderHttpTimeout int = int(30 * time.Second)
	defaultLoaderInterval    int = int(300 * time.Second)
	defaultHttpRetryCount    int = 3
	defaultHttpRetryDelay    int = int(1 * time.Second)
)
```

Reply:
I think we should first try switching to string and seeing if that works, since that's how the durations are parsed in the yaml file. We should refrain from using two conventions for time units, this will complicate our codebase. 

It still bugs me a lot that we're using the `--sparrowName` and the config.yaml value for it is simply `name`.

Reply:
Or would that require extra parsing logic when binding the flag?

Reply:
@lvlcn-t this is not working for int32 i would assume since 300s converted to int will overflow the int32

@puffitos yeah lets sw to `name` should be fine I think

Reply:
There's also a `Duration()` flag, this should solve our problems :) @NiklasTreml  you wanna give a try?

Reply:
@NiklasTreml couldn't resist and started an implementation here https://github.com/caas-team/sparrow/tree/chore/duration-flag, in case you need some inspiration ;)

Reply:
We can just set the flags type to be a `time.Duration` as @puffitos said. 
![image](https://github.com/caas-team/sparrow/assets/27763017/154cc339-7369-4c1b-983f-a9be6f9e789a)
I'll update the flags in the afternoon

Reply:
😂